stl/inc/algorithm

```diff
@@ -1017,6 +1015,34 @@ namespace ranges {
                 }
                 // If we get here, _Count > 1, initial elements do not match, and final elements do not match.
 
+                const auto _ProjectedPred = [&]<class _Ty1, class _Ty2>(_Ty1&& _Left, _Ty2&& _Right) {
+                    return _STD invoke(_Pred, _STD invoke(_Proj1, _STD forward<_Ty1>(_Left)),
+                        _STD invoke(_Proj2, _STD forward<_Ty2>(_Right)));
+                };
+
+                for (;;) {
+                    _TrimResult _Res = _Trim_reversed(_First1, _Final1, _First2, _Final2, _ProjectedPred);
+                    if (_Res != _TrimResult::_Continue) {
+                        if (_Res == _TrimResult::_Break) {
+                            break;
+                        }
+                        return _Res == _TrimResult::_ReturnTrue;
+                    }

```

Comment:
I believe it would improve readability if we would write it like a switch
```suggestion
                    
                    if (const _TrimResult _Res = _Trim_reversed(_First1, _Final1, _First2, _Final2, _ProjectedPred); _Res == _TrimResult::_Continue) {
                        // nothing
                    } else if (_Res == _TrimResult::_Break) {
                        break;
                    } else {
                        return _Res == _TrimResult::_ReturnTrue;
                    }
```

Reply:
I actually find the way it's currently written more readable.

Reply:
Ideally this would be a switch but we cannot break from it.

What I do not like is that we have the `_Res != _TrimResult::_Continue` which forces thinking about which alternatives it encompasses.

I edited it to have the assignment out of the if. I find it easier to comprehend but not a hill to die on.

Reply:
I'm ok with the sequence of `if` statements.

Reply:
I pushed a refactoring following @mnatsuhara's suggestion; I believe that this has simplified the inspection of `_Res` and avoided the potential confusion that @miscco identified.